client/cli/src/commands/mod.rs

```diff
@@ -402,6 +403,12 @@ macro_rules! substrate_cli_subcommands {
 					$($enum::$variant(cmd) => cmd.log_filters()),*
 				}
 			}
+
+			fn informant_prefix<C: SubstrateCli>(&self) -> $crate::Result<String> {
```

Comment:
Why is that a CLI feature?

I think we should move the informant out of the CLI. Aka being activated by the service or being done completely "manual". 

If we move it into the service, we could probably enable it by default as it is done currently.

Reply:
It's already in the Service's Configuration! But that struct is generated by the trait ConfigurationCli. Here it's kinda easy to customize because you can add:

```rust
fn informant_prefix() -> &'static str {
    "BOO!"
}
```

In here: https://github.com/paritytech/substrate/blob/master/bin/node/cli/src/command.rs#L24

And it works easily. It will be as easy in cumulus. :smiley:

Reply:
(So the answer is that it is a sc-service feature (makes sense) + a sc-cli feature (for convenience))

Do you have another idea?

Reply:
I'm still not convinced that this should be done in CLI. We should move it entirely to the service. Have it enabled by default and provide a function to set the prefix in the service.

Reply:
What is the reasoning to have it as a `cli` feature? I can see that for running a custom node, you might want to customize it so – as a developer. But why would a validator or anyone else running the node care to replace it?

Reply:
It's first a Service configuration setting as you can see in client/service/src/config.rs: https://github.com/paritytech/substrate/pull/6174/files#diff-fc7dfe9c2ea7011575c8b225844cd030R105

Then to set this setting you can use the CliConfiguration trait OR the SubstrateCli trait.

The only known use case for this feature for now is when you have multiple services running and you want to distinguish them in the logs.

The informant is built there: https://github.com/paritytech/substrate/pull/6174/files#diff-8f956a9810426119c8963e0eb8bd2b74R215 There aren't many alternatives to the current implementation. Maybe I could pass an extra argument to the runner... @bkchr can you detail a bit how you would do?

> But why would a validator or anyone else running the node care to replace it?

Sorry I don't know.

Reply:
If we don't have any use case (we can come up with) I'd rather not pollute the cli with params. From the comment, I understand @bkchr has an idea to structure this differently without involving the cli, which I'd appreciate.

Reply:
> I think you understand this now after the DM's?

> I understand @bkchr has an idea to structure this differently without involving the cli, which I'd appreciate.

I can move it to the service builder instead. Would that work?

Reply:
Yeah